### ISSUE TITLE
lighten and darken with amount 0 are not returning the original color

### ISSUE THREAD

**Issue by marsonmao** [ricokahler/color2k#211]
For example:

          const color = lighten('#ffb537', 0);

and

          const color = darken('#ffb537', 0);

both return `#ffb638`.

I think it should return the original color?

**Comment by ricokahler** [comment]
Yeah that's definitely a weird result. I'll take a look later <strike>today</strike> this week.

My gut says that it's an artifact of parsing to HSL and then back to hex (rounding error?). Maybe there's a way to round there that doesn't cause this.

**Comment by ricokahler** [comment]
My schedule has been pretty hectic and I haven't really investigated too deeply into this.

Quick pulse check: how important is this issue to you? Do you have a specific use case where this result is undesirable?

Thanks!
